client/components/app-promo/index.jsx

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import Gridicon from 'components/gridicon';
+
+import analytics from 'analytics';
+import store from 'store';
+import userUtils from 'lib/user/utils';
+import viewport from 'lib/viewport';
+
+export default React.createClass( {
+
+	displayName: 'AppPromo',
+
+	propTypes: {
+		location: React.PropTypes.string.isRequired
+	},
+
+	getInitialState: function() {
+		var show_promo = true;
+
+		if ( store.get( 'desktop_promo_dismissed' ) ) {
+			show_promo = false;
+		}
+
+		if ( userUtils.getLocaleSlug() !== 'en' ) {
```

Comment:
Minor optimization: we could combine these three checks into `if/else if` blocks.